### PR TITLE
Refactor clear all button on the new Alert Filters screen

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [] 

### Description

Refactored the Clear All button on the new Alert Filters screen
The reason for this is ```onRightButtonPress={() => {}}``` property looked a little confusing and inconsistent compared to the other ```FancyModalHeader``` usages

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Refactor clear all button on the new Alert Filters screen - daria

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Refactor clear all button on the new Alert Filters screen - daria)
- undefined (changes)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 09a6e10573751d6a6be76a44c9a9f25f8f0214e8
</p>

File: src/app/Scenes/SavedSearchAlert/screens/SavedSearchFilterScreen.tsx [hunk 1]
```diff
@@ -18,22 +9,63 @@ import { SavedSearchFilterPriceRangeQR } from "app/Scenes/SavedSearchAlert/Compo
...
         hideBottomDivider
         onLeftButtonPress={navigation.goBack}
-        renderRightButton={ClearAllButton}
-        onRightButtonPress={() => {}}
+        rightButtonText="Clear All"
+        rightButtonDisabled={disabled}
+        onRightButtonPress={handleClearAll}
       >
-        Filters
+        Filterssss
```
Comment (MounirDhahri) [inline src/app/Scenes/SavedSearchAlert/screens/SavedSearchFilterScreen.tsx:68]: 👀

**Review by MounirDhahri — COMMENTED** [review]
is this still needed?

File: src/app/Scenes/SavedSearchAlert/screens/SavedSearchFilterScreen.tsx (hunk 1 above)
Comment (dariakoko) [reply src/app/Scenes/SavedSearchAlert/screens/SavedSearchFilterScreen.tsx:68]: oops

**Comment by dariakoko** [comment]
@MounirDhahri it's not needed-needed. I saw the opportunity to clean up the code a little and took it. Feel free to close the PR if this change doesn't make sense to you

**Review by MounirDhahri — COMMENTED** [review]
I would suggest to keep your changes actually except for the one move the component within the screen. The reason behind that is that we loose on the built-in improvements when using a component when creating the component in the render method of the screen

**Comment by dariakoko** [comment]
Hey @MounirDhahri, I was reviewing this PR and didn't understand what you mean. Do you mind explaining your point one more time, please?

File: src/app/Scenes/SavedSearchAlert/screens/SavedSearchFilterScreen.tsx [hunk 2]
```diff
@@ -61,61 +93,3 @@ export const SavedSearchFilterScreen: React.FC<{}> = () => {
...
-        animate={{ opacity: disabled ? 0.3 : 1 }}
-        transition={{ type: "timing", duration: 200 }}
-      >
-        <Text underline color="black100">
-          Clear All
-        </Text>
-      </MotiView>
-    </Touchable>
-  )
-}
```
Comment (MounirDhahri) [inline src/app/Scenes/SavedSearchAlert/screens/SavedSearchFilterScreen.tsx:121]: There is no need to move this within the component above. We can keep it here

**Review by MounirDhahri — COMMENTED** [review]
Sure, with pleasure. I can hop on a call as well to explain it further

**Pull request closed without merging**